### DISCUSSION TITLE
How to hook into workspace:remove?

### DISCUSSION THREAD

**Discussion by lewsmith** [nrwl/nx#10946] [General]
Is there an example of hooking into the workspace:remove command? I have a file in the workspace root that's updated each time a  generator is called, I'm wondering how I can update this file when an application/library is removed using workspace:remove.

**Comment by ko22009** [comment]
up

**Comment by marko-hologram** [comment]
If you haven't already resolved this issue, you can most likely create your own custom generator where you call `workspace:remove` generator programatically and then do some logic after that has successfully finished in order to update your custom file. Of course, this would require you (and your team if you work with one) to always use the custom generator when removing a lib/app.

It seems it's suggested by the NX team to compose generators if you need to use multiple functionalities at once https://nx.dev/recipes/generators/composing-generators
